commands/deployment_strategy.go

```diff
@@ -0,0 +1,75 @@
+package commands
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/cloudfoundry-incubator/multiapps-cli-plugin/util"
+	"strconv"
+)
+
+type DeploymentStrategy interface {
+	CreateProcessBuilder() *util.ProcessBuilder
+}
+
+type DeployCommandStrategy struct {
+}
+
+func (d *DeployCommandStrategy) CreateProcessBuilder() *util.ProcessBuilder {
+	processBuilder := util.NewProcessBuilder()
+	processBuilder.ProcessType(deployCommandProcessTypeProvider{}.GetProcessType())
+	return processBuilder
+}
+
+type BlueGreenCommandStrategy struct {
+	noConfirm  bool
+}
+
+func (b *BlueGreenCommandStrategy) CreateProcessBuilder() *util.ProcessBuilder {
+	processBuilder := util.NewProcessBuilder()
+	processBuilder.ProcessType(blueGreenDeployCommandProcessTypeProvider{}.GetProcessType())
+	processBuilder.Parameter("noConfirm", strconv.FormatBool(b.noConfirm))
+	processBuilder.Parameter("keepOriginalAppNamesAfterDeploy", strconv.FormatBool(true))
+	return processBuilder
+}
+
+func NewDeploymentStrategy(options map[string]interface{}, flags *flag.FlagSet, typeProvider ProcessTypeProvider) (DeploymentStrategy, error) {
+	if _, ok := typeProvider.(*blueGreenDeployCommandProcessTypeProvider); ok {
+		return &BlueGreenCommandStrategy{GetBoolOpt(noConfirmOpt, options)}, nil
+	}
+	if err := validateStrategyFlag(flags); err != nil {
+		return nil, err
+	}
+	strategy := GetStringOpt(strategyOpt, options)
+	if strategy == "default" || strategy == "" {
+		return &DeployCommandStrategy{}, nil
+	}
+	return &BlueGreenCommandStrategy{GetBoolOpt(noConfirmOpt, options)}, nil
+}
+
+func validateStrategyFlag(flags *flag.FlagSet) error {
```

Comment:
Create custom validator here that is intended to be used with the already existing ones.

Reply:
Done